src/tests/work-item/work-item-list/displayAndUpdateWorkItemDetails.spec.js

```diff
@@ -140,32 +138,188 @@ var waitTime = 30000;
   });
 
 /* Verify that edits made to a workitem in the detail page, if cancelled, are discarded */
- it('should cancel edits to the workitem through its detail page - phone.', function() { 
-
+ it('should cancel edits to the workitem through its detail page - phone.', function() {
     /* Create a new workitem */
     var workItemTitle = "The test workitem title";
     var workItemUpdatedTitle = "The test workitem title - UPDATED";
+
+    var workItemDescription = "The test workitem description";
+    var workItemUpdatedDescription = "The test workitem description - UPDATED";
     page.clickWorkItemQuickAdd();
     page.typeQuickAddWorkItemTitle(workItemTitle);
+
     page.clickQuickAddSave().then(function() {
       expect(page.workItemTitle(page.firstWorkItem)).toBe(workItemTitle);
-  
+
       /* Fill in/update the new work item's title and details field */
       expect(page.workItemTitle(page.firstWorkItem)).toBe(workItemTitle);
-      page.workItemViewId(page.workItemByTitle(workItemTitle)).getText().then(function (text) { 
+      page.workItemViewId(page.workItemByTitle(workItemTitle)).getText().then(function (text) {
         var detailPage = page.clickWorkItemTitle(page.firstWorkItem, text);
 
         detailPage.clickWorkItemTitleEditIcon();
         detailPage.setWorkItemDetailTitle (workItemUpdatedTitle, false);
         detailPage.clickWorkItemTitleCancelIcon();
-        
+
         detailPage.clickWorkItemDetailCloseButton();
         browser.wait(until.presenceOf(page.workItemByTitle(workItemTitle)), waitTime, 'Failed to find workItemList');
         expect(page.workItemTitle(page.firstWorkItem)).toBe(workItemTitle);
+
+      });
+
+    });
+
+  });
+
+
+  /*Edit with blank title - Mobile Should show validation message*/
+    it('Edit with blank title - Mobile Should show validation message. -phone', function() {
+
+      /* Create a new workitem */
+      var workItemTitle = "The test workitem title";
+      var workItemUpdatedTitle = "               ";
+      var workItemDescription = "";
+      var workItemUpdatedDescription = " - UPDATED";
+      page.clickWorkItemQuickAdd();
+      page.typeQuickAddWorkItemTitle(workItemTitle);
+      page.clickQuickAddSave().then(function() {
+        expect(page.workItemTitle(page.firstWorkItem)).toBe(workItemTitle);
+
+        /* Fill in/update the new work item's title with blank and details field */
+        expect(page.workItemTitle(page.firstWorkItem)).toBe(workItemTitle);
+        page.workItemViewId(page.workItemByTitle(workItemTitle)).getText().then(function (text) {
+          var detailPage = page.clickWorkItemTitle(page.firstWorkItem, text);
+
+          detailPage.clickWorkItemTitleEditIcon();
+          detailPage.setWorkItemDetailTitle (workItemUpdatedTitle, false);
+          detailPage.clickWorkItemTitleSaveIcon();
+          expect((element(by.css(".clearfloat.alert.alert-danger"))).isPresent()).toBeTruthy();
```

Comment:
The current code is not incorrect, and it does run without error, but I think it needs a couple of changes.

First, we do not want any selector code in the tests. These should be all be placed in the page object model code so that the tests can be written at a higher level and will be easier to read and maintain. The "by.css" selector should be moved into the detail page object model code.

Second, while code will find the "alert-danger" UI element, it is too generic and could locate the wrong, or multiple UI elements. I'd suggest replacing this expect statement with 2 statements, each of which check for these new UI selectors: 

.//*[@id='wi-title-div'][.//*[contains(@class, 'alert-danger')]]
.//*[@id='wi-title-div']//p[.//text()[contains(.,'Title is required')]]

Yes - we'll need new code in the detail page object model for these too.   ;-)